types/wordpress__blocks/api/registration.d.ts

```diff
@@ -124,7 +124,7 @@ export function registerBlockStyle(blockName: string, styleVariation: BlockStyle
  */
 export function registerBlockType<TAttributes extends Record<string, any> = {}>(
     metadata: BlockConfiguration<TAttributes>,
-    settings?: BlockConfiguration<TAttributes>,
+    settings?: Partial<BlockConfiguration<TAttributes>>,
```

Comment:
I think we could find a way to say "you have to supply the missing metadata" here but no need in this PR. maybe a dog-ear for the future.

Reply:
I'm not entirely sure, but I think a `block.json` is expected to contain all the required properties anyway, so I don't think that would be necessary.